bid.go

```diff
@@ -19,22 +17,28 @@ var (
-	NURL       string          `json:"nurl,omitempty"`    // Win notice URL.
-	AdMarkup   string          `json:"adm,omitempty"`     // Actual ad markup. XHTML if a response to a banner object, or VAST XML if a response to a video object.
-	AdvDomain  []string        `json:"adomain,omitempty"` // Advertiser’s primary or top-level domain for advertiser checking; or multiple if imp rotating.
-	IURL       string          `json:"iurl,omitempty"`    // Sample image URL.
-	CampaignID string          `json:"cid,omitempty"`     // Campaign ID that appears with the Ad markup.
-	CreativeID string          `json:"crid,omitempty"`    // Creative ID for reporting content issues or defects. This could also be used as a reference to a creative ID that is posted with an exchange.
-	Cat        []string        `json:"cat,omitempty"`     // IAB content categories of the creative. Refer to List 5.1
-	Attr       []int           `json:"attr,omitempty"`    // Array of creative attributes.
-	DealID     string          `json:"dealid,omitempty"` // DealID extension of private marketplace deals
-	H          int             `json:"h,omitempty"`       // Height of the ad in pixels.
-	W          int             `json:"w,omitempty"`       // Width of the ad in pixels.
-	Ext        json.RawMessage `json:"ext,omitempty"`
+	ID         string   `json:"id"`
+	ImpID      string   `json:"impid"`             // Required string ID of the impression object to which this bid applies.
+	Price      float64  `json:"price"`             // Bid price in CPM. Suggests using integer math for accounting to avoid rounding errors.
+	AdID       string   `json:"adid,omitempty"`    // References the ad to be served if the bid wins.
+	NURL       string   `json:"nurl,omitempty"`    // Win notice URL.
+	AdMarkup   string   `json:"adm,omitempty"`     // Actual ad markup. XHTML if a response to a banner object, or VAST XML if a response to a video object.
+	AdvDomain  []string `json:"adomain,omitempty"` // Advertiser’s primary or top-level domain for advertiser checking; or multiple if imp rotating.
+	IURL       string   `json:"iurl,omitempty"`    // Sample image URL.
+	CampaignID string   `json:"cid,omitempty"`     // Campaign ID that appears with the Ad markup.
+	CreativeID string   `json:"crid,omitempty"`    // Creative ID for reporting content issues or defects. This could also be used as a reference to a creative ID that is posted with an exchange.
+	Cat        []string `json:"cat,omitempty"`     // IAB content categories of the creative. Refer to List 5.1
+	Attr       []int    `json:"attr,omitempty"`    // Array of creative attributes.
+	DealID     string   `json:"dealid,omitempty"`  // DealID extension of private marketplace deals
+	H          int      `json:"h,omitempty"`       // Height of the ad in pixels.
+	W          int      `json:"w,omitempty"`       // Width of the ad in pixels.
+	Ext        BidExt   `json:"ext,omitempty"`
```

Comment:
Please remove as extensions are specific for each exchange/provider. We kept them as `json.RawMessage` so that you can write your own custom encoding/decoding.